include/nil/crypto3/hash/detail/keccak/keccak_policy.hpp

```diff
@@ -48,12 +48,13 @@ namespace nil {
 
                     constexpr static const std::size_t length_bits = 0;
 
-                    typedef typename stream_endian::big_octet_big_bit digest_endian;
+                    typedef typename stream_endian::little_octet_big_bit digest_endian;
```

Comment:
I wonder if this change will interfere with something? Maybe they intentionally put big_octet_big_bit by default because our EVM works that way? You also changed the tests to match this. We only needed to change this to match some original implementation?

Reply:
I think, they put it this way to match tests input from references. And then we have to [reverse it](https://github.com/NilFoundation/crypto3-hash/pull/136/files#diff-7a1fe4312a81440fbed13629ffa172e61a561f76d491eb4ea3b136a02c285324L67) each time in permutation and digest extraction.
In current design accumulator should consume not stream of arbitrary values (otherwise, stream_processors would be redundant), but either words or blocks. SHA2 and Keccac counts bits right to left, thus, 4 bits of `0xa00000000000000b` word are actually `0xb`, not `0xa` in terms of these hashes.
As you could see in [blake2b tests](https://github.com/NilFoundation/crypto3-hash/blob/09a8b462b3c313f9b2930d59416e94a074d5990d/test/blake2b.cpp#L435), for other hashes that use little_octet endianness, it is already done this way.

Another design we could consider here is to put conversion to big_octet_big_bit (more intuitive for humans) endianness into stream_processor, so accumulator would store it in big_octet_big_bit, and then convert block to hash endianness when passing it to construction.

Reply:
Ok got it. I think you have described this to me a few days ago, I forgot about it. Let's keep it as is.